dbms/src/Flash/Mpp/MPPTunnel.h

```diff
@@ -101,122 +229,68 @@ class MPPTunnelBase : private boost::noncopyable
     // finish the writing.
     void writeDone();
 
-    std::shared_ptr<mpp::MPPDataPacket> readForLocal();
-
     /// close() finishes the tunnel, if the tunnel is connected already, it will
     /// write the error message to the tunnel, otherwise it just close the tunnel
     void close(const String & reason);
 
     // a MPPConn request has arrived. it will build connection by this tunnel;
-    void connect(Writer * writer_);
+    virtual void connect(PacketWriter * writer);
```

Comment:
Why need to be a virtual function, seems I don't found override version of it

Reply:
Good catch, it is a mistake, I'll fix it.